src/Client.php

```diff
@@ -39,7 +39,6 @@ class Client
      */
     public function __construct($config)
     {
-        $config = Config::ensure($config);
```

Comment:
@zimbatm do you mean something along the lines of:

```php
$pusherConfig = new Config('https://key:secret@api.pusherapp.com/apps/1234');
$pusher = new Pusher($pusherConfig);
```
Rather than passing all the parameters to the constructor?